src/files/FilesPage.js

```diff
@@ -1,2 +1,114 @@
 import React from 'react'
-export default () => <h1 data-id='title'>Files</h1>
+import PropTypes from 'prop-types'
+import { connect } from 'redux-bundler-react'
+import Breadcrumbs from './breadcrumbs/Breadcrumbs'
+import FilesList from './files-list/FilesList'
+import FilePreview from './file-preview/FilePreview'
+import FileInput from './file-input/FileInput'
+
+const action = (name) => {
+  return (...args) => {
+    console.log(name, args)
+  }
+}
+
+const empty = (
+  <div>
+    <h2>It seems a bit lonely here :(</h2>
+  </div>
+)
+
+class FilesPage extends React.Component {
+  static propTypes = {
+    files: PropTypes.object
+  }
+
+  state = {
+    clipboard: [],
+    copy: false
+  }
+
+  onLinkClick = (link) => {
+    const {doUpdateHash} = this.props
+    doUpdateHash(`/files${link}`)
+  }
+
+  onInspect = (hash) => {
+    const {doUpdateHash} = this.props
+    doUpdateHash(`/explore/ipfs/${hash}`)
+  }
+
+  onRename = ([path]) => {
+    const oldName = path.split('/').pop()
+    const newName = window.prompt('Insert the new name:')
+
+    if (newName) {
+      this.props.doFilesRename(path, path.replace(oldName, newName))
+    }
+  }
+
+  onFilesUpload = (files) => {
+    this.props.doFilesWrite(this.props.files.path, files)
+  }
+
+  componentWillMount () {
+    console.log('WUOLL MOUNT')
```

Comment:
WULLY

Reply:
I don't even remember why I added this listeners... 😆